app/test/features/chat/custom_input.dart

```diff
@@ -0,0 +1,222 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter/features/chat/widgets/custom_input.dart';
+import 'package:convenient_test_dev/convenient_test_dev.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../helpers/mock_chat_message.dart';
+import '../../helpers/test_wrapper_widget.dart';
+
+void main() {
+  group(
+    'Custom Chat Input - General States',
+    () {
+      testWidgets(
+        'No access, no show',
+        (tester) async {
+          await tester.pumpWidget(
+            ProviderScope(
+              overrides: [
+                // Same as before
+                canSendProvider.overrideWith((ref, roomId) => false),
+              ],
+              child: const InActerContextTestWrapper(
+                child: CustomChatInput(
+                  roomId: 'roomId',
+                ),
+              ),
+            ),
+          );
+          expect(find.byKey(CustomChatInput.noAccessKey), findsOneWidget);
+        },
+      );
+
+      testWidgets(
+        'Unknown Access show Loading',
+        (tester) async {
+          await tester.pumpWidget(
+            ProviderScope(
+              overrides: [
+                // Same as before
+                canSendProvider
+                    .overrideWith((ref, roomId) => null), // null means loading
+              ],
+              child: const InActerContextTestWrapper(
+                child: CustomChatInput(
+                  roomId: 'roomId',
+                ),
+              ),
+            ),
+          );
+          expect(find.byKey(CustomChatInput.loadingKey), findsOneWidget);
+        },
+      );
+
+      testWidgets(
+        'Show input',
+        (tester) async {
+          await tester.pumpWidget(
+            ProviderScope(
+              overrides: [
+                // Same as before
+                canSendProvider.overrideWith((ref, roomId) => true),
+                isRoomEncryptedProvider.overrideWith((ref, roomId) => true),
+              ],
+              child: const InActerContextTestWrapper(
+                child: CustomChatInput(
+                  roomId: 'roomId',
+                ),
+              ),
+            ),
+          );
+          expect(find.byKey(CustomChatInput.noAccessKey), findsNothing);
+          expect(find.byKey(CustomChatInput.loadingKey), findsNothing);
+        },
+      );
+    },
+  );
+
+  group('Custom Chat Input - Edit States', () {
+    final overrides = [
+      canSendProvider.overrideWith((ref, roomId) => true),
+      isRoomEncryptedProvider.overrideWith((ref, roomId) => true),
+    ];
+
+    testWidgets(
```

Comment:
TextEditingControllers value along with riverpod provider didn't worked for us before, so better is to listen controller which act as ```Stream``` under and use ```notifyListeners``` method. I'll add a fix and use this test to verify! 👍🏻